### ISSUE TITLE
Can't configure espresso driver to use build-tools 21.1.1

### ISSUE THREAD

**Issue by sfseyhan** [appium/appium-espresso-driver#634]
I'm trying to use expresso driver in AWS device farm where only build tools 21.1.1 and sdk 21 is pre-installed.

I'm setting buildToolsVersion capability to 21.1.1 as follows but espresso driver tries to install latest build tools and sdk fails due to lack of write permission.

capabilities.setCapability("buildToolsVersion", "21.1.1");

Pre-installed android packages that I need to work with.

Installed packages:
  Path                 | Version | Description                       | Location             
  -------              | ------- | -------                           | -------              
  build-tools;21.1.1   | 21.1.1  | Android SDK Build-Tools 21.1.1    | build-tools/21.1.1/  
  emulator             | 30.0.0  | Android Emulator                  | emulator/            
  patcher;v4           | 1       | SDK Patch Applier v4              | patcher/v4/          
  platform-tools       | 26.0.1  | Android SDK Platform-Tools 26.0.1 | platform-tools/      
  platforms;android-21 | 1       | Android SDK Platform 21           | platforms/android-21/

I have also tries using espresso capability 'espressoBuildConfig' as follows, using 1.19.0-rc3, and espresso doesn't try to install the sdk anymore but still tries to install latest build tools and doesn't use provided version 21.1.1.

capabilities.setCapability("espressoBuildConfig", "{\"toolsVersions\": {\"compileSdk\": 21,\"buildTools\": \"21.1.1\"}}");

**Comment by mykola-mokhnach** [comment]
The possibility to provide build config from capabilities as json was only added recently and is available in appium@beta

**Comment by sfseyhan** [comment]
I'm using appium server version 1.19.0-rc3 to be able to use json arguments, and it seems to accept compileSdk parameter but ignores buildTools parameter

**Comment by mykola-mokhnach** [comment]
I think recent gradle releases enforce the usage of recent build tools version and actually ignore the value provided in the config. Enable and check the actual gradle log for your build, there should be some notes about it.

**Comment by sfseyhan** [comment]
You're right. I see a warning about it with gradle logs.

"WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.0."

So I checked Android Gradle Plugin versions. AndroidGradlePlugin 3.0.0 requires a minimum build tools version 26.0.1.
https://developer.android.com/studio/releases/gradle-plugin#3-0-0

In order to use buildTools 21.1.1, latest Android Gradle Plugin I can use seems to be 2.3.3

But when I set Android Gradle Plugin version 2.3.3, I get the following error from gradle.

Kotlin: Unsupported version of com.android.tools.build:gradle plugin: version 3.0.0 or higher should be used with kotlin-android plugin.

@mykola-mokhnach Do you see any way of using Appium Espresso driver in AWS device farm where only 21.1.1 build tools is pre-installed?

**Comment by mykola-mokhnach** [comment]
Unfortunately I don't see a way to make it possible. Perhaps their customer support could help to update build tools

**Comment by sfseyhan** [comment]
I've opened an issue at aws device farm for this, in case anyone has the same issue, here's the link.

https://forums.aws.amazon.com/thread.jspa?threadID=330933

**Comment by sfseyhan** [comment]
Is there a version of appium espresso driver that is configurable with espressoBuildConfig and doesn't use kotlin-android plugin?

**Comment by KazuCocoa** [comment]
No, unfortunately. espressoBuildConfig was added after introducing Kotlin in this module
